src/App.tsx

```diff
@@ -14,11 +14,121 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  hideList: boolean,
+  isRevers: boolean,
+  sortBy: string,
+};
+
+class App extends React.Component<{}, State> {
+  state = {
+    hideList: true,
+    isRevers: false,
+    sortBy: '',
+  };
+
+  show = () => {
+    this.setState({ hideList: false });
+  };
+
+  reverse = () => {
+    this.setState(state => ({
+      isRevers: !state.isRevers,
+    }));
+  };
+
+  sortByAbc = () => {
+    this.setState({ sortBy: 'abc' });
+  };
+
+  sortByLength = () => {
+    this.setState({ sortBy: 'length' });
+  };
+
+  reset = () => {
+    this.setState({ isRevers: false, sortBy: '' });
+  };
```

Comment:
instead of sorting goods in `render()` try adding a single function for sort with a switch/case inside where you would sort the goods and in render just show them